### PR TITLE
0.8 patching

### PR DESCRIPTION
**Pull request by sorvell**
Adds additional accessors to Polymer.dom: first/last child/ElementChild, previous/next sibling/ElementSibling, textContent, innerHTML

Adds experimental `patch-dom.html` import which patches dom that interacts with Shady DOM such that normal dom tree traversal and mutation api's can be used. This can enhance interoperation with 3rd party code that is not aware of Polymer.dom.

### CODE REVIEW

File: src/lib/dom-api.html [hunk 1]
```diff
@@ -370,18 +351,170 @@
...
+          configurable: true  
+        },
+
+        // textContent / innerHTML
+        textContent: {
+          get: function() {
+            if (this.node.nodeType === Node.TEXT_NODE) {
+              return this.node.textContent;
+            } else {
+              return Array.prototype.map.call(this.childNodes, function(c) {
```
Comment (kevinpschaaf) [inline src/lib/dom-api.html]: `this.childNodes` -> `Polymer.dom(this).childNodes`

File: src/lib/dom-api.html [hunk 2]
```diff
@@ -370,18 +351,170 @@
...
+          get: function() {
+            var c$ = this.children;
+            return c$[c$.length-1];
+          },
+          configurable: true  
+        },
+
+        nextElementSibling: {
+          get: function() {
+            var c$ = this.parentNode && factory(this.parentNode).children;
```
Comment (kevinpschaaf) [inline src/lib/dom-api.html]: Polymer.dom here too

File: src/lib/dom-api.html [hunk 3]
```diff
@@ -402,6 +535,107 @@
...
+          get: function() {
+            return this.node.innerHTML;
+          },
+          set: function(value) {
+            return this.node.innerHTML = value;
+          },
+          configurable: true  
+        }
+
+      });
```
Comment (kevinpschaaf) [inline src/lib/dom-api.html]: Factory-ize the native wrappers to reduce MLOCs

File: src/lib/expr/patch-dom.html [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+    accessors: ['parentNode', 'childNodes', 
+      'firstChild', 'lastChild', 'nextSibling', 'previousSibling',
+      'parentElement', 'children',
+      'firstElementChild', 'lastElementChild', 
+      'nextElementSibling', 'previousElementSibling'
+    ],
+
+    preAccessors: ['children', 'parentElement',
+      'firstChild', 'lastChild', 'nextSibling', 'previousSibling',
```
Comment (kevinpschaaf) [inline src/lib/expr/patch-dom.html]: not used; remove

File: src/mini/shady.html [hunk 5]
```diff
@@ -349,6 +350,13 @@
     var nativeRemoveChild = Element.prototype.removeChild;
 
     function insertBefore(parentNode, newChild, refChild) {
+      if (newChild.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
+        console.log(parentNode, newChild, refChild);
```
Comment (kevinpschaaf) [inline src/mini/shady.html]: remove logging

File: src/lib/dom-api.html (hunk 1 above)
Comment (sorvell) [inline src/lib/dom-api.html]: The calling context is Polymer.dom, so this is correct.

**Comment by sorvell** [comment]
@kevinpschaaf PTAL

**Comment by kevinpschaaf** [comment]
LGTM

**Pull request merged**